clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -191,27 +226,47 @@ mlir::Value lowerCirAttrAsValue(mlir::Operation *parentOp,
                                 mlir::ConversionPatternRewriter &rewriter,
                                 mlir::TypeConverter *converter) {
   auto module = parentOp->getParentOfType<mlir::ModuleOp>();
-  auto sourceSymbol = dyn_cast<mlir::LLVM::GlobalOp>(
-      mlir::SymbolTable::lookupSymbolIn(module, globalAttr.getSymbol()));
-  assert(sourceSymbol && "Unlowered GlobalOp");
-  auto loc = parentOp->getLoc();
-
-  auto addressOfOp = rewriter.create<mlir::LLVM::AddressOfOp>(
-      loc, mlir::LLVM::LLVMPointerType::get(sourceSymbol.getType()),
-      sourceSymbol.getSymName());
+  mlir::Type sourceType;
+  llvm::StringRef symName;
+  auto sourceSymbol = mlir::SymbolTable::lookupSymbolIn(module, globalAttr.getSymbol());
+  if (auto llvmSymbol = dyn_cast<mlir::LLVM::GlobalOp>(sourceSymbol)) {
+    sourceType = llvmSymbol.getType();
+    symName = llvmSymbol.getSymName();
+  } else if (auto cirSymbol = dyn_cast<mlir::cir::GlobalOp>(sourceSymbol)) {
```

Comment:
Do we have a test for this case?

Reply:
Yes, `_ZTI1B` in vbase.cpp. Note that it's used in `_ZTV1B` before defined.